### PR TITLE
build: upgrade transitive dependencies to minor and modernize build

### PR DESCRIPTION
**Pull request by sdelamo**
- grpc  1.50.2
- protobuf 3.21.8
- protobuf-gradle 0.9.1
-  jackson-datatype-protobuf 0.9.13

### CODE REVIEW

**Comment by sdelamo** [comment]
Port to branch compatible with Micronaut Framework 3 changes from @altro3  at  https://github.com/micronaut-projects/micronaut-grpc/pull/606

File: gradle.properties [hunk 1]
```diff
@@ -1,13 +1,9 @@
...
-micronautTestVersion=3.5.0
-
-micronautGradlePluginVersion=3.5.1
-protobufGradleVersion=0.8.19
-
-groovyVersion=3.0.12
+micronautVersion=3.7.3
+micronautTestVersion=3.7.0
+groovyVersion=3.0.13
 spockVersion=2.1-groovy-3.0
```
Comment (timyates) [inline gradle.properties:7]: `2.3-groovy-3.0`?

File: buildSrc/src/main/groovy/io.micronaut.build.internal.grpc-base.gradle [hunk 2]
```diff
@@ -0,0 +1,16 @@
+repositories {
+    mavenCentral()
+    maven {
+        url "https://s01.oss.sonatype.org/content/repositories/snapshots/"
+        mavenContent {
+            snapshotsOnly()
+        }
+    }
```
Comment (timyates) [inline buildSrc/src/main/groovy/io.micronaut.build.internal.grpc-base.gradle:8]: Why do we need the snapshot repo?

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -1,55 +1,42 @@
...
-micronaut-inject-java = { module = 'io.micronaut:micronaut-inject-java' }
-micronaut-management = { module = 'io.micronaut:micronaut-management' }
-micronaut-reactor-http-client = { module = 'io.micronaut.reactor:micronaut-reactor-http-client' }
-micronaut-runtime = { module = 'io.micronaut:micronaut-runtime' }
-micronaut-tracing = { module = 'io.micronaut:micronaut-tracing' }
-micronaut-validation = { module = 'io.micronaut:micronaut-validation' }
+managed-protoc = { module = 'com.google.protobuf:protoc', version.ref = 'managed-protobuf' }
 
-graal = { module = "org.graalvm.nativeimage:svm" }
+micronaut-docs-asciidoc-config-props = { module = 'io.micronaut.docs:micronaut-docs-asciidoc-config-props', version.ref = 'micronaut-docs' }
```
Comment (timyates) [inline gradle/libs.versions.toml:23]: Do we need this?  I thought it came with the internal build?

**Review by timyates — APPROVED** [review]
👍 a few comments

File: buildSrc/src/main/groovy/io.micronaut.build.internal.grpc-base.gradle (hunk 2 above)
Comment (sdelamo) [reply buildSrc/src/main/groovy/io.micronaut.build.internal.grpc-base.gradle:8]: we will need once we target 4.0.0-SNAPShOT

File: gradle/libs.versions.toml (hunk 3 above)
Comment (sdelamo) [reply gradle/libs.versions.toml]: resolved via https://github.com/micronaut-projects/micronaut-grpc/pull/606/commits/731368939b3ea41c349d4da8f6ef1ebf0d8f1a9a

File: gradle.properties (hunk 1 above)
Comment (sdelamo) [reply gradle.properties]: resolved via https://github.com/micronaut-projects/micronaut-grpc/pull/606/commits/26896b840c8d4bc0065199b557942b92bebd6a0d

**Pull request merged**